pyro/distributions/omt_mvn.py

```diff
@@ -36,18 +36,16 @@ def rsample(self, sample_shape=torch.Size()):
 class _OMTMVNSample(Function):
     @staticmethod
     def forward(ctx, loc, scale_tril, shape):
-        ctx.white = loc.new_empty(shape).normal_()
-        ctx.z = torch.matmul(ctx.white, scale_tril.t())
-        ctx.save_for_backward(scale_tril)
-        return loc + ctx.z
+        white = loc.new_empty(shape).normal_()
+        z = torch.matmul(white, scale_tril.t())
+        ctx.save_for_backward(z, white, scale_tril)
```

Comment:
Does this work? I thought you could only save inputs or outputs for backward. (That was certainly the case in PyTorch 0.3)

Reply:
i wouldn't push a proposed fix if i hadn't tested it

Reply:
Can we confirm that PyTorch changed behavior before merging this? I.e. locate the PyTorch PR where this behavior changed, or locate discussion about it?

Reply:
is adnotation #3 helpful?

https://github.com/pytorch/pytorch/pull/1016

Reply:
It looks like the behavior may have changed in https://github.com/pytorch/pytorch/pull/4786/files so I guess this is fine.